src/components/Personalization/in-app-message-actions/actions/displayIframeContent.js

```diff
@@ -0,0 +1,310 @@
+    width: "100%",
+    height: "100%",
+    background: "transparent",
+    opacity,
+    backgroundColor: color
+  };
+  return style;
+};
+
+const REQUIRED_PARAMS = ["enabled", "parentElement", "insertionMethod"];
+
+const isValidWebParameters = webParameters => {
+  if (!webParameters) {
+    return false;
+  }
+
+  const ids = Object.keys(webParameters);
+
+  if (!includes(ids, ALLOY_MESSAGING_CONTAINER_ID)) {
+    return false;
+  }
+
+  if (!includes(ids, ALLOY_OVERLAY_CONTAINER_ID)) {
+    return false;
+  }
+
+  const valuesArray = values(webParameters);
+
+  for (let i = 0; i < valuesArray.length; i += 1) {
+    if (!objectOf(valuesArray[i], "style")) {
+      return false;
+    }
+
+    if (!objectOf(valuesArray[i], "params")) {
+      return false;
+    }
+
+    for (let j = 0; j < REQUIRED_PARAMS.length; j += 1) {
+      if (!objectOf(valuesArray[i].params, REQUIRED_PARAMS[j])) {
+        return false;
+      }
+    }
+  }
+
+  return true;
+};
+
+const generateWebParameters = mobileParameters => {
+  if (!mobileParameters) {
+    return undefined;
+  }
+
+  const { uiTakeover = false } = mobileParameters;
+
+  return {
+    [ALLOY_IFRAME_ID]: {
+      style: {
+        border: "none",
+        width: "100%",
+        height: "100%"
+      },
+      params: {
+        enabled: true,
+        parentElement: "#alloy-messaging-container",
+        insertionMethod: "appendChild"
+      }
+    },
+    [ALLOY_MESSAGING_CONTAINER_ID]: {
+      style: buildStyleFromMobileParameters(mobileParameters),
+      params: {
+        enabled: true,
+        parentElement: "body",
+        insertionMethod: "appendChild"
+      }
+    },
+    [ALLOY_OVERLAY_CONTAINER_ID]: {
+      style: mobileOverlay(mobileParameters),
+      params: {
+        enabled: uiTakeover === true,
+        parentElement: "body",
+        insertionMethod: "appendChild"
+      }
+    }
+  };
+};
+
+export const displayHTMLContentInIframe = (settings = {}, interact) => {
+  dismissMessage();
+  const { content, contentType, mobileParameters } = settings;
+  let { webParameters } = settings;
+
+  if (contentType !== TEXT_HTML) {
+    return;
+  }
+
+  const container = createNode("div", { id: ALLOY_MESSAGING_CONTAINER_ID });
+  const iframe = createIframe(content, createIframeClickHandler(interact));
+  const overlay = createNode("div", { id: ALLOY_OVERLAY_CONTAINER_ID });
+
+  if (!isValidWebParameters(webParameters)) {
+    webParameters = generateWebParameters(mobileParameters);
+  }
+
+  if (!webParameters) {
+    return;
+  }
+
+  renderMessage(iframe, webParameters, container, overlay);
+};
+
+export default (settings, collect) => {
+  return new Promise(resolve => {
+    const { meta } = settings;
+    displayHTMLContentInIframe(settings, (action, propositionAction) => {
+      const propositionEventTypes = new Set();
+      propositionEventTypes.add(PropositionEventType.INTERACT);
+
+      if (Object.values(PropositionEventType).indexOf(action) !== -1) {
```

Comment:
please extract into a descriptive function

Reply:
fixed in https://github.com/adobe/alloy/pull/1070